jetcd-core/src/main/java/com/coreos/jetcd/internal/impl/WatchImpl.java

```diff
@@ -94,54 +97,50 @@ public Watcher watch(ByteSequence key) {
 
   @Override
   public synchronized Watcher watch(ByteSequence key, WatchOption watchOption) {
-    if (isClosed()) {
+    if (this.isClosed()) {
       throw newClosedWatchClientException();
     }
 
     WatcherImpl watcher = new WatcherImpl(key, watchOption, this);
     this.pendingWatchers.add(watcher);
-
-    if (this.pendingWatchers.size() == 1) {
-      // head of the queue send watchCreate request.
-      WatchRequest request = this.toWatchCreateRequest(watcher);
-      this.getGrpcWatchStreamObserver().onNext(request);
-    }
+    this.triggerWatcherCreation();
```

Comment:
what's the issue with previous implementation on 

```java
if (this.pendingWatchers.size() == 1) {		
 -      // head of the queue send watchCreate request.		
 -      WatchRequest request = this.toWatchCreateRequest(watcher);		
 -      this.getGrpcWatchStreamObserver().onNext(request);		
 -    }
```

Reply:
the issue was that with the retry logic and multiple open watchers we got concurrency issues.